src/Microsoft.DotNet.Cli.Utils/CommandResolution/PublishedPathCommandResolver.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    public class PublishedPathCommandResolver : ICommandResolver
```

Comment:
It won't work for a built app, because it does not binplace your package dependencies. In which case, we would not be able to find commands, like dotnet-test-xunit. They only appear when you publish the app. Does that make sense?